src/Webhooks/DeliveryMethod.php

```diff
@@ -23,51 +26,83 @@ abstract public function getCallbackAddress(string $path): string;
      *
-     * @return string
+     * @return string GraphQL query string to check if a topic is already registered.
      */
     abstract public function buildCheckQuery(string $topic): string;
 
     /**
-     * @param array $body
+     * Parses the result of the check query and returns the webhookId and current delivery address.
+     * 
+     * This method interprets the response to extract meaningful data, like the webhook ID and its delivery address.
+     *
+     * @param array $body The response body from the GraphQL query.
      *
-     * @return array Array of the webhookId and current delivery address
+     * @return array Array containing the webhookId and current delivery address.
      */
     abstract public function parseCheckQueryResult(array $body): array;
 
     /**
-     * Assembles a GraphQL query for registering a webhook.
+     * Assembles a GraphQL query for registering or updating a webhook subscription.
+     * 
+     * This method now supports adding additional optional fields and metafield namespaces,
+     * which allows further customization of the webhook subscription.
+     * 
+     * The operation (create/update) is determined by the webhookId.
      *
-     * @param string      $topic
-     * @param string      $callbackAddress
-     * @param string|null $webhookId
+     * @param string      $topic               The topic for the webhook subscription.
+     * @param string      $callbackAddress     The callback URL for the webhook.
+     * @param string|null $webhookId           Optional webhook ID for updating an existing subscription.
+     * @param array       $fields              Optional fields to include in the webhook subscription.
+     * @param array       $metafieldNamespaces Optional metafield namespaces to include.
      *
-     * @return string
+     * @return string The GraphQL query to register or update the webhook.
      */
     public function buildRegisterQuery(
         string $topic,
         string $callbackAddress,
-        ?string $webhookId = null
+        ?string $webhookId = null,
+        array $fields = [],
+        array $metafieldNamespaces = []
     ): string {
+        // Use the mutation name based on the presence of the webhookId
         $mutationName = $this->getMutationName($webhookId);
-        $identifier = $webhookId ? "id: \"$webhookId\"" : "topic: $topic";
+
+        // If updating, use the webhookId; if creating, use the topic
+        $identifier = $webhookId ? "id: \"$webhookId\"" : "topic: \"$topic\"";
+
+        // Prepare optional fields and metafield namespaces
+        $fieldsQuery = !empty($fields) ? 'fields: [' . implode(',', $fields) . ']' : '';
+        $metafieldNamespacesQuery = !empty($metafieldNamespaces) ? 'metafieldNamespaces: [' . implode(',', $metafieldNamespaces) . ']' : '';
+
+        // Assemble the webhook subscription arguments
         $webhookSubscriptionArgs = $this->queryEndpoint($callbackAddress);
```

Comment:
We need to use this method to properly build the callback URL, for each of the webhook cases Pubsub, HTTP, eventbridge.